### PR TITLE
Add recipes for Mender 2.3.1 components to rocko

### PR DESCRIPTION
**Pull request by lluiscampos**
Supersedes #1075

### CODE REVIEW

**Comment by lluiscampos** [comment]
Testing @ [Jenkins 4483](https://mender-jenkins.mender.io/job/mender-builder/4483/)

**Comment by lluiscampos** [comment]
Pushed a test fix to deprecate Artifact format v1 and running pipeline @ [Jenkins 4484](https://mender-jenkins.mender.io/job/mender-builder/4484/)

File: tests/acceptance/test_mender-artifact.py [hunk 1]
```diff
@@ -31,7 +31,7 @@
 LAST_BUILD_VERSION = None
 
 # params is the versions we will test.
-@pytest.fixture(scope="function", params=[1, 2, 3])
+@pytest.fixture(scope="function", params=[2, 3])
```
Comment (kacf) [inline tests/acceptance/test_mender-artifact.py:34]: I haven't looked closely at the exact changes you made, but you could consider skipping or deleting this test, as this is already tested extensively on other branches, and is almost solely dependent on the mender-artifact branch, not the meta-mender branch.

**Review by kacf — APPROVED** [review]

**Review by drewmoseley — APPROVED** [review]
Looks good to me.

@lluiscampos how did you resolve the build failures using older golang?

File: tests/acceptance/test_mender-artifact.py (hunk 1 above)
Comment (lluiscampos) [reply tests/acceptance/test_mender-artifact.py:34]: I just cherry-picked a commit because it was easier that editing the tests :)

**Comment by lluiscampos** [comment]
> Pushed a test fix to deprecate Artifact format v1 and running pipeline @ [Jenkins 4484](https://mender-jenkins.mender.io/job/mender-builder/4484/)

Summary of the Jenkins run:
* acceptance tests and integration tests passed for qemux86_64_uefi_grub (the only variant of `qemux86_64` that we support in `rocko).
* acceptance tests errored/failed for `vexpress` (for both `vexpress` variants supported `rocko`). The errors are in different tests but all related to SSH connections from paramiko.
* raspberrypi job didn't build, gave an error in `image-full-cmdline-1.0-r0 do_rootfs`

I think that acceptance and integration test pass for qemux86_64 is all the evidence that we need. @kacf do you think otherwise?

> @lluiscampos how did you resolve the build failures using older golang?

@drewmoseley Don't get confused, I am bringing to rocko Mender [product] 2.3.1, which means Mender client 2.2.1 and mender-artifact 3.3.1. Therefore avoiding the golang problems you orignaly had when trying Mender client 2.3 (Mender product 2.4).

**Comment by kacf** [comment]
The RPi failure is a bit discomforting, but I doubt it has anything to do with this pull request. More likely they broke it upstream.

So I agree that all evidence looks either positive, unrelated, or inconclusive, so for this old branch we should just move on.

**Pull request merged**